app/src/main/java/com/orgzly/android/ChooserShareTargetService.kt

```diff
@@ -0,0 +1,34 @@
+package com.orgzly.android
+
+import android.app.PendingIntent
+import android.content.ComponentName
+import android.content.IntentFilter
+import android.graphics.drawable.Icon
+import android.os.Build
+import android.service.chooser.ChooserTarget
+import android.service.chooser.ChooserTargetService
+import android.support.annotation.RequiresApi
+import kotlinx.android.synthetic.main.activity_bookchooser.view.*
+import android.graphics.BitmapFactory
+import android.os.Bundle
+import com.orgzly.R
+
+
+@RequiresApi(Build.VERSION_CODES.M)
+class ChooserShareTargetService : ChooserTargetService() {
+    private val shelf = Shelf(this)
+
+    override fun onGetChooserTargets(componentName: ComponentName?, intentFilter: IntentFilter?): MutableList<ChooserTarget> {
+        val targets = arrayListOf<ChooserTarget>();
+
+        val icon = Icon.createWithResource(this, R.drawable.ic_logo_for_widget);
+        for (book in shelf.books) {
+            val bundle : Bundle = Bundle();
+            val score : Float = 1.0F; // Can we guess a score based on recent use or content?
+            bundle.putString(AppIntent.EXTRA_FILTER, "b.\"" + book.name + "\"");
```

Comment:
This can be a problem if notebook contains quotes. Perhaps (not tested):

```suggestion
            val query = DottedQueryBuilder().build(Query(Condition.InBook(book.name)))
            bundle.putString(AppIntent.EXTRA_FILTER, query)
```

Although I don't see a reason to pass a query, maybe just a book name would be enough?

Reply:
Ah, building the query properly looks nicer.

I only used a query because the `ShareActivity` already looks for a query in the `EXTRA_FILTER`:
```
                if (intent.hasExtra(AppIntent.EXTRA_FILTER)) {
                    Query query = new DottedQueryParser().parse(intent.getStringExtra(AppIntent.EXTRA_FILTER));
                    String bookName = QueryUtils.extractFirstBookNameFromQuery(query.getCondition());
...
```

Is there a simpler way to open a book with just a name?

Reply:
> I only used a query because the ShareActivity already looks for a query in the EXTRA_FILTER

Ah right, `ShareActivity` is used for this too.